x/emissions/abci.go

```diff
@@ -1,30 +1,32 @@
 package emissions
 
 import (
+	sdkmath "cosmossdk.io/math"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/zeta-chain/zetacore/cmd/zetacored/config"
 	"github.com/zeta-chain/zetacore/x/emissions/keeper"
 	"github.com/zeta-chain/zetacore/x/emissions/types"
+	"sort"
 )
 
-func BeginBlocker(ctx sdk.Context, keeper keeper.Keeper, stakingKeeper types.StakingKeeper, bankKeeper types.BankKeeper, observerKeeper types.ZetaObserverKeeper) {
-	reservesFactor, bondFactor, durationFactor := GetBlockRewardComponents(ctx, bankKeeper, stakingKeeper, keeper)
+func BeginBlocker(ctx sdk.Context, keeper keeper.Keeper) {
+	reservesFactor, bondFactor, durationFactor := keeper.GetBlockRewardComponents(ctx)
 	blockRewards := reservesFactor.Mul(bondFactor).Mul(durationFactor)
 	if blockRewards.IsZero() {
 		return
 	}
 	validatorRewards := sdk.MustNewDecFromStr(keeper.GetParams(ctx).ValidatorEmissionPercentage).Mul(blockRewards).TruncateInt()
 	observerRewards := sdk.MustNewDecFromStr(keeper.GetParams(ctx).ObserverEmissionPercentage).Mul(blockRewards).TruncateInt()
 	tssSignerRewards := sdk.MustNewDecFromStr(keeper.GetParams(ctx).TssSignerEmissionPercentage).Mul(blockRewards).TruncateInt()
-	err := DistributeValidatorRewards(ctx, validatorRewards, bankKeeper, keeper.GetFeeCollector())
+	err := DistributeValidatorRewards(ctx, validatorRewards, keeper.GetBankKeeper(), keeper.GetFeeCollector())
 	if err != nil {
 		panic(err)
 	}
-	err = DistributeObserverRewards(ctx, observerRewards, bankKeeper, observerKeeper)
+	err = DistributeObserverRewards(ctx, observerRewards, keeper)
 	if err != nil {
 		panic(err)
 	}
-	err = DistributeTssRewards(ctx, tssSignerRewards, bankKeeper)
+	err = DistributeTssRewards(ctx, tssSignerRewards, keeper.GetBankKeeper())
```

Comment:
That is correct; a panic here would cause the chain to halt.
As far as I remember, the panic was added as a placeholder until we decide what to do here. 
Added a issue to remove it before mainnet : https://github.com/zeta-chain/node/issues/941